packages/sit-onyx/src/components/TestInput/TestInput.vue

```diff
@@ -1,16 +1,20 @@
 <script lang="ts">
 export const INPUT_TYPES = ["email", "number", "password", "search", "tel", "text", "url"] as const;
+export type InputType = (typeof INPUT_TYPES)[number];
+
+/**
+ * Input types that have a translation for their validation error message.
+ * Must be up to date with i18n/locales/en-US.json.
+ */
+const TRANSLATED_INPUT_TYPES = ["email", "number", "tel", "url"] as const satisfies InputType[];
```

Comment:
didn't we have once an implementation (in the blueprint?) where we got the typing info (= which keys exist) directly out of a translation json file? or does this not work without async?

mayby this might be less manual synching in the long run.

but if it makes it too complicated it's also fine to keep it as-is :)

Reply:
Found a nice way to do it :)